lib/logging/appenders/rolling_file.rb

```diff
@@ -84,61 +84,48 @@ def initialize( name, opts = {} )
 
       # grab our options
       @size = opts.getopt(:size, :as => Integer)
-
-      code = 'def sufficiently_aged?() false end'
+      @age = opts.getopt(:age)
       @age_fn = @fn + '.age'
       @age_fn_mtime = nil
 
-      case @age = opts.getopt(:age)
-      when 'daily'
-        code = <<-CODE
-        def sufficiently_aged?
-          @age_fn_mtime ||= ::File.mtime(@age_fn)
+      do_age_predicate_setup = lambda do
+
+        set_mtime = lambda { @age_fn_mtime ||= ::File.mtime(@age_fn) }
+
+        daily_predicate = lambda do
           now = Time.now
-          if (now.day != @age_fn_mtime.day) or (now - @age_fn_mtime) > 86400
-            return true
-          end
-          false
+          set_mtime.call
+          (now.day != @age_fn_mtime.day) or (now - @age_fn_mtime) > 86400
         end
-        CODE
-      when 'weekly'
-        code = <<-CODE
-        def sufficiently_aged?
-          @age_fn_mtime ||= ::File.mtime(@age_fn)
-          if (Time.now - @age_fn_mtime) > 604800
-            return true
-          end
-          false
+
+        weekly_predicate = lambda do
+          set_mtime.call
+          (Time.now - @age_fn_mtime) > 604800
         end
-        CODE
-      when 'monthly'
-        code = <<-CODE
-        def sufficiently_aged?
-          @age_fn_mtime ||= ::File.mtime(@age_fn)
+
+        monthly_predicate = lambda do
           now = Time.now
-          if (now.month != @age_fn_mtime.month) or (now - @age_fn_mtime) > 2678400
-            return true
-          end
-          false
+          set_mtime.call
+          (now.month != @age_fn_mtime.month) or (now - @age_fn_mtime) > 2678400
         end
-        CODE
-      when Integer, String
-        @age = Integer(@age)
-        code = <<-CODE
-        def sufficiently_aged?
-          @age_fn_mtime ||= ::File.mtime(@age_fn)
-          if (Time.now - @age_fn_mtime) > @age
-            return true
-          end
-          false
+
+        seconds_predicate = lambda do
+          set_mtime.call
+          (Time.now - @age_fn_mtime) > @age
+        end
+
+        @age_predicate = case @age
+           when 'daily';         daily_predicate
+           when 'weekly';        weekly_predicate
+           when 'monthly';       monthly_predicate
+           when Integer, String; @age = Integer(@age); seconds_predicate
+           else lambda { false }
         end
-        CODE
       end
 
-      FileUtils.touch(@age_fn) if @age and !test(?f, @age_fn)
+      do_age_predicate_setup.call
```

Comment:
It seems a bit overkill here to define a lambda and then immediately `call` the lambda after it is defined. It would be more straightforward to define a `setup_age_checks` method and invoke that from the initializer.